src/linear_proof.rs

```diff
@@ -165,7 +168,12 @@ impl LinearProof {
         b_vec: Vec<Scalar>,
     ) -> Result<(), ProofError> {
         transcript.innerproduct_domain_sep(n as u64);
+        assert_eq!(b_vec.len(), n);
```

Comment:
Not an issue with this PR, but I do not see an equivalent assertion for `G` (my expectation that its length should be `n` is due to the presence of this assertion in `LinearProof::create`).

Additionally, given that `n` here must be equal to `b_vec.len()`, it might make more sense to simply omit the `n` parameter and derive it from `b_vec` or `G`. (`LinearProof::create` does this, deriving `n` from `a_vec`).

Nit: these assertions should probably instead be returned as cases of `ProofError` to avoid DoS issues in applications using the library.